knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go

```diff
@@ -200,6 +196,23 @@ func TestKnativeKafkaReconcile(t *testing.T) {
 	}
 }
 
+func getPodTemplateSpec(cl client.WithWatch, d types.NamespacedName) (client.Object, *corev1.PodTemplateSpec, error) {
+	deployment := &appsv1.Deployment{}
+	err := cl.Get(context.TODO(), d, deployment)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, nil, err
+	}
+	if apierrors.IsNotFound(err) {
+		ss := &appsv1.StatefulSet{}
+		err = cl.Get(context.TODO(), d, ss)
+		if err != nil {
+			return nil, nil, err
+		}
+		return ss, &ss.Spec.Template, nil
+	}
+	return deployment, &deployment.Spec.Template, nil
+}
+
```

Comment:
Can you improve this function?

* It returns too many things.
* The first 2 thing it returns are related

I see this function is used 4 times. However, 

* In 2, only the returned error is used
* In 1, only the returned object and error are used
* In 1, only the returned podTemplate spec and error are used

I know this is a *test-only helper function*  but it would be nice to improve it if you can.

Reply:
what do you suggest?

I wanted to avoid duplicating the logic "get pod template of deployment or statefulset"